### PR TITLE
use SCMP_ACT_KILL_PROCESS for SystemCallFilters=

### PR DESCRIPTION
**Pull request by poettering**
A fix for #11967

### CODE REVIEW

File: src/nspawn/nspawn-oci.c [hunk 1]
```diff
@@ -1655,13 +1655,19 @@ static int oci_seccomp_action_from_string(const char *name, uint32_t *ret) {
...
-                { "SCMP_ACT_ALLOW", SCMP_ACT_ALLOW        },
-                { "SCMP_ACT_ERRNO", SCMP_ACT_ERRNO(EPERM) }, /* the OCI spec doesn't document the error, but it appears EPERM is supposed to be used */
-                { "SCMP_ACT_KILL",  SCMP_ACT_KILL         },
+                { "SCMP_ACT_ALLOW",         SCMP_ACT_ALLOW        },
+                { "SCMP_ACT_ERRNO",         SCMP_ACT_ERRNO(EPERM) }, /* the OCI spec doesn't document the error, but it appears EPERM is supposed to be used */
+                { "SCMP_ACT_KILL",          SCMP_ACT_KILL         },
+#ifdef SCMP_ACT_KILL_PROCESS
+                { "SCMP_ACT_KILL_PROCESS",  SCMP_ACT_KILL_PROCESS },
+#endif
+#ifdef SCP_ACT_KILL_THREAD
```
Comment (dnicolodi) [inline src/nspawn/nspawn-oci.c]: Typo

Comment (poettering) [reply src/nspawn/nspawn-oci.c]: indeed! thanks!

**Comment by poettering** [comment]
Force pushed a new version with the typo @dnicolodi found fixed!

File: src/shared/seccomp-util.c [hunk 2]
```diff
@@ -1964,3 +1964,18 @@ int seccomp_restrict_suid_sgid(void) {
 
         return 0;
 }
+
+uint32_t scmp_act_kill_process(void) {
+
+        /* Returns SCMP_ACT_KILL_PROCESS if it's supported, and SCMP_ACT_KILL_THREAD otherwise. We never
+         * actually want to use SCMP_ACT_KILL_THREAD as its semantics are nuts (killing arbitrary threads of
+         * unexpected programs is just a bad idea), but on old kernels/old libseccomp is all we have, and at
```
Comment (keszybz) [inline src/shared/seccomp-util.c]: unexpected programs → a program
libseccomp is → libseccomp this is

File: NEWS [hunk 3]
```diff
@@ -2,6 +2,23 @@ systemd System and Service Manager
 
 CHANGES WITH 243 in spe:
 
+        * Previously, filters defined with SystemCallFilter= would have the
+          effect that an calling an offending system call would terminate the
+          calling thread. This behaviour never made much sense, since killing
+          individual threads of unexpecting processes is likely to create more
+          problems than it solves. With this release the default action changed
+          from killing the thread to killing the whole process, which should be
```
Comment (keszybz) [inline NEWS]: ", which should be less problematic" → just drop this part. (Killing the whole process is not problematic at all, it just does the job.)

**Comment by poettering** [comment]
fixed the comment/doc issues. will change to regular green label hence.

**Pull request merged**